### PR TITLE
New element and application templates

### PR DESCRIPTION
**Pull request by justinfagnani**

### CODE REVIEW

**Comment by azakus** [comment]
PTAL @FredKSchott for yeoman stuff.

File: package.json [hunk 1]
```diff
@@ -64,6 +64,8 @@
     "depcheck": "^0.6.3",
     "eslint": "^2.9.0",
     "gulp-mocha": "^2.2.0",
+    "gulp-typescript": "^2.13.4",
+    "gulp-typings": "^1.3.6",
     "sinon": "^1.17.4",
     "temp": "^0.8.3",
     "typescript": "^1.8.10",
```
Comment (FredKSchott) [inline package.json]: get rid of typescript & typings now that we're all in on gulp?

File: src/init/application/application.ts [hunk 2]
```diff
@@ -0,0 +1,100 @@
...
+  }
+
+  prompting() {
+    var _this = this;
+
+    var prompts = [
+      {
+        name: 'name',
+        type: 'input',
+        message: `Application name (${this.appname})`,
```
Comment (FredKSchott) [inline src/init/application/application.ts]: IIRC yeoman already displays the default value next to the message in parenthesis, so including it here will result in a double-print.

File: src/init/application/application.ts [hunk 3]
```diff
@@ -0,0 +1,100 @@
...
+        default() {
```
Comment (FredKSchott) [inline src/init/application/application.ts]: nit: function is unnecessary here. if possible all defaults should be strings, or functions.

File: src/init/application/application.ts (hunk 2 above)
Comment (FredKSchott) [inline src/init/application/application.ts]: yup just confirmed

**Comment by FredKSchott** [comment]
Really excited to see this land! 
btw comments apply to both generators

File: package.json (hunk 1 above)
Comment (justinfagnani) [inline package.json]: done

File: src/init/application/application.ts (hunk 3 above)
Comment (justinfagnani) [inline src/init/application/application.ts]: done

**Comment by justinfagnani** [comment]
@FredKSchott PTAL

**Comment by FredKSchott** [comment]
LGTM!

**Pull request merged**